### PR TITLE
Feat/153 update profile screen

### LINKED ISSUE #153: update Profile screen
- Update Figma for profile screen
- Clean up Profile UI
- Show user display name and email
- Add possibility to switch App Colour Themes

### PR DESCRIPTION
**Pull request by alextheprophecy**
# Profile Screen Enhancements & App Theme toggling

## Summary
This PR introduces an updated profile screen design and implements theme customization options, improving the visual design and user control over themes in the app.

---

### 1. Profile Screen Redesign
- Updated the Figma design to provide a modern, Android-centric look for the profile screen.
- Made design updates in `ProfileScreen.kt` to align with the new Figma style. (incl. padding, alignement, colours, fontSize)
- Displayed Google account details (`displayName` and `email`) to give users a personalized profile view.

### 2. Theme Selection Dropdown
- Added a dropdown menu in the profile screen using `DropdownFields`, allowing users to choose different app themes

### 3. Theme Management Enhancements in `Theme.kt`
- Defined an enum `ThemeMode` with three options: `LIGHT`, `DARK`, `SYSTEM_DEFAULT`
  
- Defined some CompositionLocals to provide a toggleTheme function and themeMode value to be passed into the ShelfLife App. (See ShelfLifeTheme Component)
  - Created a `LocalThemeToggler` object, which is the skeleton to contain a `toggleTheme` function for handling theme changes.
  - Established `LocalThemeTogglerProvider` and `LocalThemeMode` using `compositionLocalOf`

### 4. ShelfLifeTheme big overhaul
- **Persistent Theme Mode**: The theme mode is loaded from and saved to shared preferences (`theme_mode`), ensuring the selected theme persists across app sessions, so the user does not have to change their preferences every time. It is set to SYSTEM_DEFAULT by default
- **Toggle Theme Integration**: The `toggleTheme` function updates the theme mode in real-time and saves it to shared preferences, so users' selections are maintained.
- **CompositionLocalProvider**: This provides `LocalThemeToggler` and `LocalThemeMode` within `ShelfLifeTheme`, making the theme toggler and mode accessible throughout the composition. (See `ProfileScreen.kt` for how these are accessed)

### 5. ProfileScreenTests added
- Profile tests were added to check correct display of user information 
- Test added to check theme modes inside the dorpdown menu are clickable

---

## Testing Notes (For Reviewer)
- Check that components adapt correctly to theme changes.
- Check that chosen theme is saved, after closing app.
- Test the app on devices with different Android versions to confirm compatibility, especially for dynamic color support and system theme alignment.

---

## Important Notes
- The theme toggle inside the Profile screen navigates back to startingRoute inside `MainActivity`. This was done because of some visual bugs, where the Theme was getting correctly updated, but the UI was not changing accordingly.
- > TLDR: profile theme toggling moves navigation to Overview Screen
- Code coverage: Achieved exactly 80% LINE code coverage on new code inside Theme.kt, since sharedPreferences was tricky to mock. However ProfileScreenTest has 96% coverage (Overall PR has 91% code coverage)

### CODE REVIEW

File: app/src/androidTest/java/com/android/shelflife/ui/profile/ProfileScreenTest.kt [hunk 1]
```diff
@@ -9,30 +10,52 @@ import androidx.test.ext.junit.runners.AndroidJUnit4
...
+import org.mockito.Mockito.`when`
 
 @RunWith(AndroidJUnit4::class)
 class ProfileScreenTest {
 
   @get:Rule val composeTestRule = createComposeRule()
 
   @Test
-  fun testGreetingText_whenAccountIsNull_displaysGuest() {
+  fun testNameText_whenAccountIsNull_displaysGuest() {
```
Comment (AleMerRom) [inline app/src/androidTest/java/com/android/shelflife/ui/profile/ProfileScreenTest.kt:34]: good you have changed the name to somthing more representative.

PS : We need to decide a format for test function names

File: app/src/main/java/com/android/shelfLife/MainActivity.kt [hunk 2]
```diff
@@ -76,6 +80,14 @@ fun ShelfLifeApp() {
     HouseholdViewModel(HouseholdRepositoryFirestore(firebaseFirestore), listFoodItemViewModel)
   }
```
Comment (AleMerRom) [inline app/src/main/java/com/android/shelfLife/MainActivity.kt:82]: very nice way to solve a bug

File: app/src/main/java/com/android/shelfLife/ui/profile/ProfileScreen.kt [hunk 3]
```diff
@@ -73,11 +97,44 @@ fun ProfileScreen(
...
+                  text = currentAccount?.displayName ?: "Guest",
                   fontWeight = FontWeight.Bold,
+                  color = color.primary,
                   fontSize = 28.sp,
-                  textAlign = TextAlign.Left,
-                  modifier = Modifier.testTag("greetingText"))
+                  textAlign = TextAlign.Center,
+                  modifier = Modifier.testTag("profileNameText"))
+
+              if (currentAccount?.email != null) {
```
Comment (AleMerRom) [inline app/src/main/java/com/android/shelfLife/ui/profile/ProfileScreen.kt:107]: you place an if, should we add an else (even if it is impossible to reach this point) just in case?

File: app/src/main/java/com/android/shelfLife/ui/profile/ProfileScreen.kt [hunk 4]
```diff
@@ -73,11 +97,44 @@ fun ProfileScreen(
               }
               Spacer(modifier = Modifier.height(32.dp))
               Text(
-                  text = "Hello, ${currentAccount?.email ?: "Guest"}",
+                  text = currentAccount?.displayName ?: "Guest",
```
Comment (AleMerRom) [inline app/src/main/java/com/android/shelfLife/ui/profile/ProfileScreen.kt:100]: I really like that you post the name of the user. It feels more personalized

**Review by AleMerRom — APPROVED** [review]
### **PR Review**
---
### **Summary**
Thank you for solving issue #153 and contributing to the project. You have vastly improved the profile screen. 

---
What's new ? 
--
**Functionality**
1 . Redesign of the profile screen
- I really like that you have a more personalized profile screen. the fact that the image 
of the email account appears is very nice, and the name of the user is also a nice touch. The email address is still there, but is no longer the most important component in the screen which is a score. 
 - You have added a DropDownMenu with the options of themes. This is brilliant. I have tested on my phone multiple times and all the functionalities worked. More on it later. My only criticism on the DDM is the fact that is filled in. It has a gray background that I am not really a fan. We could discuss it with the rest of the team and maybe change it in the future ( it is purely esthetic)
- the rest of functions still work, i.e. the log out button

2. Theme.kt
- Using an Enum for the themes really enhances the management 
- State Persistence with Shared Preferences:
   - You have made it possible for the app to remember the preferences of the user. This means that the theme will stay until they change it
- Dynamic Theme Toggle Functionality:
  - the logic for selecting colorTheme based on the themeMode and the capabilities of the device is very nicely implemented

3. MainActivity
   - I really like your fix to the bug you were having. It shows you have very good problem solving skills

**Code Quality:**
- Organization and Structure:

   - The code is well-organized, with clear separation of concerns across different files and components.
   - The naming conventions are consistent and descriptive, making the codebase easier to navigate. We do need to meet up to discuss the naming conventions for test functions

**Tests**
You have added tests and modified pre-existing ones. This means the coverage of this PR is at 91,5% line coverage (Very nice)

Potential corner case:
- In the profile screen, you check  that the email is not null but do not check the else option. Realistically no one is reaching that point without the email account, but just in case, you should do something if such a situation occurs.

Overall, a very good job

Keep up with the good work
